### PR TITLE
[getopt-win32] ARM64 support

### PR DESCRIPTION
**Pull request by ZacWalk**
**Describe the pull request**

- #### What does your PR fix?
  This PR adds arm64-windows  support for getopt-win32

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  yes - removed arm64-windows

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes
  
  Several ports were not explicit in not supporting ARM64: graphviz, ideviceinstaller, libirecovery. They had getopt-win32 dependency that stopped them failing ARM64 CI. I have changed relevant "supports" attributes to explicitly not support ARM64. I did not increment port versions as there were no changes to actual libraries.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/libirecovery/vcpkg.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version-string": "1.0.25",
   "port-version": 3,
   "description": "Library and utility to talk to iBoot/iBSS via USB on Mac OS X, Windows, and Linux",
+  "supports": "!(windows & arm)",
```
Comment (Adela0814) [inline ports/libirecovery/vcpkg.json:6]: Should this be `!(windows & arm64)`. Port `ideviceinstaller `and `graphviz `are the same.

In fact, I think we need to confirm whether these ports really do not support arm64. If there is only fail in CI due to an issue, we can add it to `ci.baseline` and fix it later.

Comment (Adela0814) [inline ports/libirecovery/vcpkg.json:6]: I don't think the `supports `field needs to be added here

File: ports/getopt-win32/portfile.cmake [hunk 2]
```diff
@@ -8,7 +8,8 @@ vcpkg_from_github(
     REF 0.1
     SHA512 40e2a901241a5d751cec741e5de423c8f19b105572c7cae18adb6e69be0b408efc6c9a2ecaeb62f117745eac0d093f30d6b91d88c1a27e1f7be91f0e84fdf199
     HEAD_REF master
-    PATCHES getopt.h.patch
+    PATCHES getopt.h.patch 
+		arm64.patch
```
Comment (Adela0814) [inline ports/getopt-win32/portfile.cmake:12]:
```suggestion
    PATCHES getopt.h.patch 
		arm64.patch
```suggestion
    PATCHES 
        getopt.h.patch 
        arm64.patch

```
```

Comment (Adela0814) [inline ports/getopt-win32/portfile.cmake:12]:
```suggestion
    PATCHES getopt.h.patch 
		arm64.patch
```suggestion
    PATCHES 
        getopt.h.patch 
        arm64.patch

```
```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: versions/g-/graphviz.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "c1f20d0e1aaccb1035e3fe3eb95d005b4161a56e",
```
Comment (ras0219-msft) [inline versions/g-/graphviz.json:4]: Modifying existing `git-tree` entries in the version database is never correct. Any changes to the ports requires incrementing the `port-version` and publishing a new port.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR!

I noticed that the patch was much, much longer than a simple CMake build, so I've pushed a change to use a replacement buildsystem.

Could you please address the one outstanding comment I have? LGTM after that.

**Comment by ZacWalk** [comment]
Thanks @ras0219-msft  and @Adela0814.  Using CMAKE is a better approach. Hopefully everything is OK now.

**Comment by ras0219-msft** [comment]
Yep, LGTM. Thanks!

**Pull request merged**